ios/GenericClient.swift

```diff
@@ -15,67 +15,101 @@ class GenericClient: NSObject {
+        let parameters = options["body"] == JSON.null ? nil : options["body"]
+        let encoder: ParameterEncoder = parameters != nil ? JSONParameterEncoder.default : URLEncodedFormParameterEncoder.default
+        let headers = getHTTPHeadersFrom(options: options)
+        let requestModifer = getRequestModifierFrom(options: options)
+        let interceptor = getInterceptorFrom(options: options)
 
-        AF.request(url, method: .delete, parameters: parameters, encoder: encoder, headers: headers).responseJSON { response in
-            resolve(["headers": response.response?.allHeaderFields, "data": response.value])
+        AF.request(url, method: method, parameters: parameters, encoder: encoder, headers: headers, interceptor: interceptor, requestModifier: requestModifer).responseJSON { json in
+            resolve([
+                "headers": json.response?.allHeaderFields,
+                "data": json.value,
+                "code": json.response?.statusCode,
+                "lastRequestedUrl": json.response?.url?.absoluteString
+            ])
         }
     }
 
-    func optionsHeadersToHTTPHeaders(headers: Dictionary<String, String>) -> HTTPHeaders {
-        var httpHeaders = HTTPHeaders()
-        for (name, value) in headers {
-            httpHeaders.add(name: name, value: value)
+    func getInterceptorFrom(options: JSON) -> Interceptor? {
+        let adapters = getRequestAdaptersFrom(options: options)
+        let retriers = getRequestRetriersFrom(options: options)
+
+        if (!adapters.isEmpty && !retriers.isEmpty) {
+            return Interceptor(adapters: adapters, retriers: retriers)
+        }
+        if (!adapters.isEmpty) {
+            return Interceptor(adapters: adapters)
+        }
+        if (!retriers.isEmpty) {
+            return Interceptor(retriers: retriers)
+        }
+        
+        return nil
+    }
+    
+    func getRequestAdaptersFrom(options: JSON) -> [RequestAdapter] {
+        let adapters = [RequestAdapter]()
+        
+        return adapters
+    }
+
+    func getRequestRetriersFrom(options: JSON) -> [RequestRetrier] {
+        var retriers = [RequestRetrier]()
+
+        let configuration = options["retryPolicyConfiguration"]
+        if configuration["type"].string == EXPONENTIAL_RETRY {
+            let retryLimit = configuration["retryLimit"].uInt ?? RetryPolicy.defaultRetryLimit
+            let exponentialBackoffBase = configuration["exponentialBackoffBase"].uInt ?? RetryPolicy.defaultExponentialBackoffBase
+            let exponentialBackoffScale = configuration["exponentialBackoffScale"].double ?? RetryPolicy.defaultExponentialBackoffScale
+
+            let retryPolicy = RetryPolicy(retryLimit: retryLimit, exponentialBackoffBase: exponentialBackoffBase, exponentialBackoffScale: exponentialBackoffScale)
+            retriers.append(retryPolicy)
+        }
+
+        return retriers
+    }
```

Comment:
this code is being duplicated, can you extract it to a different file or something ?